ksqldb-engine/src/test/java/io/confluent/ksql/analyzer/QueryValidatorUtilTest.java

```diff
@@ -0,0 +1,67 @@
+package io.confluent.ksql.analyzer;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import io.confluent.ksql.metastore.model.DataSource;
+import io.confluent.ksql.schema.ksql.Column;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.SystemColumns;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QueryValidatorUtilTest {
+
+  @Mock
+  private Analysis analysis;
+  @Mock
+  private Analysis.AliasedDataSource aliasedDataSource;
+  @Mock
+  private DataSource dataSource;
+  @Mock
+  private LogicalSchema logicalSchema;
+  @Mock
+  private Column column;
+  @Mock
+  private KsqlConfig ksqlConfig;
+
+
+  @Test
+  public void shouldThrowOnUserColumnsWithSameNameAsNewPseudoColumns() {
```

Comment:
Let's add an analogous test into each of PullQueryValidatorTest and PushQueryValidatorTest, since what we really want to test is that pull and push queries are disallowed. (Currently there's a test gap in that we never validate that PullQueryValidator and PushQueryValidator actually call this new util method.

We can remove the test from here once we've done that, or we can leave it.

Reply:
Wouldn't a better way of doing this be to just mock the call to this with the mocked Analysis to throw? This way the functionality of `QueryValidatorUtil` is still tested by the corresponding `Test` file, but we also check to make sure `{Push/Pull}QueryValidator` are calling the method?